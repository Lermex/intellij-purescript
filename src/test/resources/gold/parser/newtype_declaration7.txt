Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('Foo')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('Foo')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PsiElement(()('(')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PSASTWrapperElement(TypeConstructor)
                    PsiElement(proper name)('String')
                    PsiElement(whitespace)(' ')
            PsiElement(->)('->')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Type)
              PSASTWrapperElement(TypeAtom)
                PsiElement(()('(')
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PSASTWrapperElement(TypeConstructor)
                          PsiElement(proper name)('Array')
                          PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PSASTWrapperElement(TypeConstructor)
                          PsiElement(proper name)('User')
                PsiElement())(')')
          PsiElement())(')')