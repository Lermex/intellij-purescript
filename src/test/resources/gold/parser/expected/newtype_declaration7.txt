Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Foo')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Foo')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PsiElement(()('(')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PsiElement(proper name)('String')
                  PsiElement(whitespace)(' ')
            PsiElement(->)('->')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Type)
              PSASTWrapperElement(TypeAtom)
                PsiElement(()('(')
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('Array')
                        PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('User')
                PsiElement())(')')
          PsiElement())(')')